lib/datasource/gitlab-tags/index.ts

```diff
@@ -69,3 +68,64 @@ export async function getReleases({
   );
   return dependency;
 }
+
+/**
+ * gitlab.getDigest
+ *
+ * This function will simply return the latest commit hash for the configured repository.
+ */
+export async function getDigest(
+  { lookupName: repo, registryUrl }: Partial<DigestConfig>,
+  newValue?: string
+): Promise<string | null> {
+  const depHost = getDepHost(registryUrl);
+
+  const cachedResult = await packageCache.get<string>(
+    cacheNamespace,
+    getCacheKey(depHost, repo, 'commit')
+  );
+  // istanbul ignore if
+  if (cachedResult) {
+    return cachedResult;
+  }
+
+  const urlEncodedRepo = encodeURIComponent(repo);
+  let digest;
+
+  if (newValue) {
+    const url = joinUrlParts(
+      depHost,
+      `api/v4/projects`,
+      urlEncodedRepo,
+      `repository/commits/`,
+      newValue
+    );
+    const gitlabCommits = await gitlabApi.getJson<GitlabCommit>(url, {
+      paginate: true,
+    });
+    digest = gitlabCommits.body.id;
+  } else {
+    const url = joinUrlParts(
+      depHost,
+      `api/v4/projects`,
+      urlEncodedRepo,
+      `repository/commits?per_page=1`
+    );
+    const gitlabCommits = await gitlabApi.getJson<GitlabCommit[]>(url, {
+      paginate: true,
```

Comment:
I think we only need the first page? You're now fetching all commits one-by-one.